test/widgets/compose_box_test.dart

```diff
@@ -1239,4 +1245,41 @@ void main() {
         maxHeight: verticalPadding + 170 * 1.5, maxVisibleLines: 6);
     });
   });
+
+  group('ComposeBoxState new-event-queue transition', () {
+    testWidgets('content input not cleared when store changes', (tester) async {
+      // Regression test for: https://github.com/zulip/zulip-flutter/issues/1470
+
+      TypingNotifier.debugEnable = false;
+      addTearDown(TypingNotifier.debugReset);
+
+      final channel = eg.stream();
+      await prepareComposeBox(tester,
+        narrow: eg.topicNarrow(channel.streamId, 'topic'), streams: [channel]);
+
+      await enterContent(tester, 'some content');
+      checkContentInputValue(tester, 'some content');
+
+      store.updateMachine!
+        ..debugPauseLoop()
+        ..poll()
+        ..debugPrepareLoopError(
+            eg.apiExceptionBadEventQueueId(queueId: store.queueId))
+        ..debugAdvanceLoop();
+      await tester.pump();
+
+      // new store replaces the old one
+      final newStore = testBinding.globalStore.perAccountSync(eg.selfAccount.id)!;
```

Comment:
How about this?

```suggestion
      final newStore = testBinding.globalStore.perAccountSync(store.accountId)!;
```

This way we don't rely on `prepareComposeBox` using a specific value for self-account ID.

Reply:
Ooh, thanks for thinking of this.